### PR TITLE
Update tifffile for 2017.01.12 changes

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: skimage/external/tifffile/tifffile.c [hunk 1]
```diff
@@ -10,13 +10,14 @@ Refer to the tifffile.py module for documentation and tests.
...
-:Version: 2016.04.13
+:Version: 2017.01.10
 
 Requirements
 ------------
 * `CPython 2.7 or 3.5 <http://www.python.org>`_
-* `Numpy 1.10 <http://www.numpy.org>`_
+* `Numpy 1.11 <http://www.numpy.org>`_
 * A Python distutils compatible C compiler  (build)
+* `stdint.h <https://github.com/chemeris/msinttypes/>`_ for msvc9 compiler
```
Comment (blink1073) [inline skimage/external/tifffile/tifffile.c]: It looks like we need to include these headers on Appveyor, but I have no idea how to do that.

Comment (blink1073) [inline skimage/external/tifffile/tifffile.c]: Success!

**Comment by sciunto** [comment]
Are the modifications in c and py files coming from upstream?

You choose 0.14 as a release target. Any reason to do not merge for 0.13?

**Comment by sciunto** [comment]
your readme file https://github.com/blink1073/tifffile looks outdated (if I understand well), on pypi as well. :)

**Comment by blink1073** [comment]
Yes, these are from http://www.lfd.uci.edu/~gohlke/code/tifffile.py.html with only slight modifications to make the doctests pass (we don't have some of the source files referenced).

**Comment by blink1073** [comment]
0.13 works as well, didn't want to impose anything else on that milestone.

**Comment by blink1073** [comment]
Thanks for the heads up about the readme, updating.

**Comment by sciunto** [comment]
Here are the revision notes for this update:

```
2017.01.12
    Read Zeiss SEM metadata.
    Read OME-TIFF with invalid references to external files.
    Rewrite C LZW decoder (5x faster).
    Read corrupted LSM files missing EOI code in LZW stream.
2017.01.01
    Add option to append images to existing TIFF files.
    Read files without pages.
    Read S-FEG and Helios NanoLab tags created by FEI software.
    Allow saving Color Filter Array (CFA) images.
    Add info functions returning more information about TiffFile and TiffPage.
    Add option to read specific pages only.
    Remove maxpages argument (backwards incompatible).
    Remove test_tifffile function.
2016.10.28
    Pass 1944 tests.
    Improve detection of ImageJ hyperstacks.
    Read TVIPS metadata created by EM-MENU (by Marco Oster).
    Add option to disable using OME-XML metadata.
    Allow non-integer range attributes in modulo tags (by Stuart Berg).
2016.06.21
    Do not always memmap contiguous data in page series.
2016.05.13
    Add option to specify resolution unit.
    Write grayscale images with extra samples when planarconfig is specified.
    Do not write RGB color images with 2 samples.
    Reorder TiffWriter.save keyword arguments (backwards incompatible).
2016.04.18
    Pass 1932 tests.
    TiffWriter, imread, and imsave accept open binary file streams.

```

**Comment by sciunto** [comment]
As there is speed and support enhancement without code review to do on our side, I would be +1 to include that in 0.13... @jni ?

**Comment by soupault** [comment]
R.I.P. AppVeyor (see https://github.com/scikit-image/scikit-image/pull/2507, https://ci.appveyor.com/project/soupault/scikit-image/build/1.0.1343)

**Pull request merged**